File: frontend/tekprovider-ui/src/types/customer.ts
export type CustomerStatus = 'Active' | 'Inactive' | 'Suspended';

export interface Customer {
  id: string;
  name: string;
  taxId: string;
  email: string;
  phone: string | null;
  status: CustomerStatus;
  createdAt: string;
  rowVersion: number;
}

export interface PagedResult<T> {
  items: T[];
  totalCount: number;
  page: number;
  pageSize: number;
}

export interface CreateCustomerInput {
  name: string;
  taxId: string;
  email: string;
  phone?: string | null;
}

export interface UpdateCustomerInput extends CreateCustomerInput {
  id: string;
  rowVersion: number;
}

export interface ChangeCustomerStatusInput {
  id: string;
  newStatus: CustomerStatus;
}

// Mirrors the seeded CustomerStatusTransitions matrix in TekProvider.Infrastructure — keep in sync.
export const CUSTOMER_STATUS_TRANSITIONS: Record<CustomerStatus, CustomerStatus[]> = {
  Active: ['Inactive', 'Suspended'],
  Inactive: ['Active'],
  Suspended: ['Active'],
};

File: BackEnd/TekProvider.Domain/Entities/Customer.cs
using TekProvider.Domain.Enums;
using TekProvider.Domain.Exceptions;

namespace TekProvider.Domain.Entities;

public sealed class Customer
{
    public Guid Id { get; private set; }
    public string Name { get; private set; } = string.Empty;
    public string TaxId { get; private set; } = string.Empty;
    public string Email { get; private set; } = string.Empty;
    public string? Phone { get; private set; }
    public CustomerStatus Status { get; private set; }
    public DateTime CreatedAt { get; private set; }
    public uint RowVersion { get; private set; }

    private Customer()
    {
    }

    public static Customer Create(string name, string taxId, string email, string? phone)
    {
        return new Customer
        {
            Id = Guid.NewGuid(),
            Name = name,
            TaxId = taxId,
            Email = email,
            Phone = phone,
            Status = CustomerStatus.Active,
            CreatedAt = DateTime.UtcNow
        };
    }

    public void Update(string name, string taxId, string email, string? phone)
    {
        Name = name;
        TaxId = taxId;
        Email = email;
        Phone = phone;
    }

    public void ChangeStatus(CustomerStatus newStatus, bool transitionAllowed)
    {
        if (!transitionAllowed)
        {
            throw new InvalidStateTransitionException(Status, newStatus);
        }

        Status = newStatus;
    }
}
